Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -33,28 +33,204 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
+           
+            let combinedImage = getCombinedAvatarImage(from: images) ?? appearance.images.userAvatarPlaceholder2
+            
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: combinedImage,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        loadAvatarsFrom(urls: urls, channelId: channel.cid) { [weak self] avatars, channelId in
+            guard let self = self, channelId == self.content.channel?.cid else { return }
+            
+            let combinedImage = self.getCombinedAvatarImage(from: avatars) ?? self.appearance.images.userAvatarPlaceholder2
+            
+            self.presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: combinedImage,
+                preferredSize: .avatarThumbnailSize,
+                components: self.components
+            )
+        }
+    }
+    
+    private func loadAvatarsFrom(
+        urls: [URL?],
+        channelId: ChannelId,
+        completion: @escaping ([UIImage], ChannelId)
+            -> Void
+    ) {
+        let group = DispatchGroup()
+        var avatars: [UIImage] = []
+        let placeholders = appearance.images.avatarPlaceholders
+        
+        for urlIndex in 0..<urls.count {
+            if let imageUrl = urls[urlIndex] {
+                group.enter()
+                ImageLoader.loader.loadImage(from: imageUrl, components: components) { result in
+                    switch result {
+                    case let .success(image):
+                        avatars.append(image)
+                    case let .failure(failure):
+                        avatars.append(placeholders[urlIndex])
+                        log.error(failure.localizedDescription)
+                    }
+                    group.leave()
+                }
+            } else {
+                avatars.append(appearance.images.avatarPlaceholders[urlIndex])
             }
-
-            // TODO: https://stream-io.atlassian.net/browse/CIS-652
-            // this is just placeholder implementation:
-
-            let firstOtherMember = channel.lastActiveMembers
-                .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
-                .first(where: { $0.id != content.currentUserId })
-
-            return (firstOtherMember?.imageURL, firstOtherMember?.isOnline ?? false)
-        }()
-
-        presenceAvatarView.avatarView.imageView.loadImage(
-            from: avatarURL,
-            placeholder: appearance.images.userAvatarPlaceholder4,
-            preferredSize: .avatarThumbnailSize,
-            components: components
-        )
-        presenceAvatarView.isOnlineIndicatorVisible = isOnlineIndicatorVisible
+        }
+        
+        group.notify(queue: DispatchQueue.main) {
+            completion(avatars, channelId)
+        }
```

Comment:
This should be part of `ImageLoader` component. To have something like `ImageLoader.loadAllImages(urls:, placeholders:, completion:)`.

Reply:
As discussed, loading images for an array of URLs doesn't seem to be a responsibility of the loader, and it also creates a problem with the parameters even if we try to do so, hence skipping this and keeping the method in the avatarview itself